internal/server/httpapi/trigger.go

```diff
@@ -0,0 +1,378 @@
+package httpapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gorilla/mux"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint/internal/clicontext"
+	"github.com/hashicorp/waypoint/internal/serverclient"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/hashicorp/waypoint/pkg/serverconfig"
+)
+
+// Message is the message we return to the requester when streaming output
+type Message struct {
+	JobId     string      `json:"jobId,omitempty"`
+	Value     interface{} `json:"value,omitempty"`
+	ValueType string      `json:"valueType,omitempty"`
+	ExitCode  string      `json:"exitCode,omitempty"`
+	Error     string      `json:"error,omitempty"`
+}
+
+// HandleTrigger will execute a run trigger, if the requested id exists
+// This works by connecting back to our own local gRPC server.
+func HandleTrigger(addr string, tls bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		log := hclog.FromContext(ctx)
+		log.SetLevel(hclog.Debug)
+
+		// Authless trigger URLs should be able to make a request
+		// without a token.
+		token := r.URL.Query().Get("token")
+		requireAuth := true
+		if token == "" {
+			log.Trace("no token provided, will attempt to run authless trigger")
+			requireAuth = false
+		}
+
+		// Connect back to our own gRPC service.
+		grpcConn, err := serverclient.Connect(ctx,
+			serverclient.Logger(log),
+			serverclient.FromContextConfig(&clicontext.Config{
+				Server: serverconfig.Client{
+					Address:     addr,
+					RequireAuth: requireAuth,
+					AuthToken:   token,
+
+					// Our gRPC server should always be listening on TLS.
+					// We ignore it because its coming out of our own process.
+					Tls:           tls,
+					TlsSkipVerify: true,
+				},
+			}),
+		)
+		if err != nil {
+			log.Error("trigger connection back to gRPC failed", "err", err)
+			return
+		}
+		defer grpcConn.Close()
+
+		// Our API client
+		client := pb.NewWaypointClient(grpcConn)
+
+		requestVars := mux.Vars(r)
+		runTriggerId := requestVars["id"]
+
+		overrideVarJSONRaw := r.URL.Query().Get("variables")
+		var (
+			vo                map[string]string
```

Comment:
Nit pick on names, suggest renaming:

- `overrideVarJSONRaw` -> `variableJSONRaw`
- `vo` -> `variables`

I feel like that will line up with the `Get("variables")` part above better. At first read it wasn't clear what `vo` represented, or would represent, especially sitting just above the `variableOverrides` variable.